src/detectors/FHCAL/FHCAL.cc

```diff
@@ -69,7 +69,7 @@ extern "C" {
           "HcalEndcapPInsertMergedHits", {"HcalEndcapPInsertRecHits"}, {"HcalEndcapPInsertMergedHits"},
           {
             .readout = "HcalEndcapPInsertHits",
-            .fields = {"layer", "slice"},
+            .fields = {"layer", "slice", "side"},
```

Comment:
We could check the readout for presence of the side field to support the older geometry.

Reply:
Right now doing it like it's done here is easy because EICrecon is lax with exceptions (they are not fatal, only insert would be affected), but we'd want to move to a more strict behavior. Whatever preparation can be done upfront, we should try to do.